plugins/modules/nd_federation_member.py

```diff
@@ -0,0 +1,280 @@
+    - M(cisco.nd.nd_federation_member) can only be used with python 3.7 and higher.
+    - Can only be used with Nexus Dashboard versions >= 2.3(2d).
+    - The ND version of the local cluster must be greater than or equal to the ND versions of the clusters being added.
+author:
+    - Anvitha Jain (@anvjain)
+options:
+  clusters:
+    description:
+      - The IP address of the cluster.
+    elements: dict
+    suboptions:
+      cluster_hostname:
+        description:
+          - The IP address of the federation member/cluster.
+        type: str
+        aliases: [ cluster_ip, hostname, ip_address, federation_member ]
+      cluster_username:
+        description:
+          - The username for the federation member/cluster.
+        type: str
+      cluster_password:
+        description:
+          - The password for the federation member/cluster.
+        type: str
+        no_log: true
+      cluster_login_domain:
+        description:
+          - The login domain ame to use for the federation member/cluster.
+          - Default value is set to DefaultAuth.
+        type: str
+        default: "DefaultAuth"
+  state:
+    description:
+      - The state of the cluster configuration.
+    type: str
+    default: present
+    choices: [ absent, present, query ]
+extends_documentation_fragment: cisco.nd.modules
+notes:
+- The M(cisco.aci.nd_federation) module can be used for this.
+"""
+
+EXAMPLES = r"""
+- name: Setup multi-cluster configuration
+  cisco.nd.nd_federation_member:
+    host: nd
+    username: admin
+    password: SomeSecretPassword
+    clusters:
+      - cluster_hostname: 172.37.20.15
+        cluster_username: admin
+        cluster_password: password
+        cluster_login_domain: default
+    state: present
+  delegate_to: localhost
+
+- name: Query all federation members
+  cisco.nd.nd_federation_member:
+    host: nd
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+
+- name: Query a federation member
+  cisco.nd.nd_federation_member:
+    host: nd
+    username: admin
+    password: SomeSecretPassword
+    clusters:
+      - cluster_hostname: 172.37.20.15
+    state: query
+  delegate_to: localhost
+
+- name: Remove a federation member
+  cisco.nd.nd_federation_member:
+    host: nd
+    username: admin
+    password: SomeSecretPassword
+    cluster: 172.37.20.15
+    state: absent
+  delegate_to: localhost
+"""
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        clusters=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                cluster_hostname=dict(type="str", aliases=["cluster_ip", "hostname", "ip_address", "federation_member"]),
```

Comment:
I am confused about your cluster_hostname. Is this not a required parameter? I see it is required for query, and present. So whenever you specify a cluster in clusters the cluster_hostname is required I think.

This would simplify the logic in lines 146-151

Reply:
changed